lib/api/models/pokemon_type_model.dart

```diff
@@ -8,7 +8,7 @@ part 'pokemon_type_model.g.dart';
 class PokemonType with _$PokemonType {
   factory PokemonType({
     @JsonKey(name: 'id') int? id,
-    @JsonKey(name: 'types') List<SubType>? types,
+    @JsonKey(name: 'types') List<SubType>? subType,
```

Comment:
please make the name property inside the JsonKey identical to your variable name